scripts/script.js

```diff
@@ -8,17 +8,34 @@ const groups = [
     'c_draw',
 ];
 
+function stars(rating){
+
+    let ret = [];
+    for(let i = 1; i <= 5; i++){
+        if(rating >= i){
+            ret.push('<i class="fa-solid fa-star"></i>');
+        }
+        else if(rating+0.5>=i){
+            ret.push('<i class="fa-solid fa-star-half-stroke"></i>');
+        }
+        else{
+            ret.push('<i class="fa-regular fa-star"></i>');
+        }
+    }
+    return ret.join('\n');
+}
+
 function load_courses(group, filter_s=''){
     
-    fetch(`./data/${group.slice(2)}_res.json`)
```

Comment:
You should fetch from the server not from the file directly also in the future you may want to add .env file so you can have different links for each environment (production -developing) read more [here](https://www.freecodecamp.org/news/how-to-use-environment-variables-in-vanillajs/)